packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/cap.dart

```diff
@@ -44,11 +73,36 @@ class Cap {
     double refWidth = 10,
   }) {
     assert(refWidth > 0.0);
-    return Cap._(<Object>['customCap', bitmapDescriptor.toJson(), refWidth]);
+    return CustomCap(bitmapDescriptor, refWidth: refWidth);
   }
 
-  final Object _json;
+  /// The type of rendering used for the cap at a start or end vertex of a
+  /// [Polyline].
+  final CapType capType;
```

Comment:
Sorry, I missed this before; this should just be called `type`. The fact that it's a cap type is inherent in the enclosing class.